### PR TITLE
GitHubCI: add a job to check commitlint status for PRs

### PR DESCRIPTION
**Pull request by knocte**
This also includes 2 custom plugins that I developed really quickly.

~Tested locally, but didn't test PR CI yet.~

### CODE REVIEW

**Comment by knocte** [comment]
@aarani I'm gonna use NOnion as guinea pig for this experiment if you don't mind :) On Monday can you:
1) test it.
2) convert the .js file to typescript and change from tabs to spaces pls 🙏

**Comment by knocte** [comment]
> On Monday can you:
> 1. test it.
> 2. convert the .js file to typescript and change from tabs to spaces pls 🙏

In the end I tested it and converted it to TS. But I won't merge this until we get unit tests for the 2 plugins.

**Comment by knocte** [comment]
@aarani please review

File: .github/workflows/commitlint.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: commitlint check
```
Comment (aarani) [inline .github/workflows/commitlint.yml:1]: A better workflow name would be nice (one that describes what it does, smth like `Lint commit msgs`)

Comment (aarani) [reply .github/workflows/commitlint.yml:1]: Because this is shown in the Action tab

**Comment by aarani** [comment]
Beside my comment and request to please merge it with squash, looks good to me.

**Comment by knocte** [comment]
> please merge it with squash

If I do that we lose the interesting info I wrote down in the commit msgs.

**Pull request merged**